packages/blade/src/components/Card/CardRoot.web.tsx

```diff
@@ -37,8 +37,12 @@ const StyledCardRoot = styled(BaseBox)<CardRootProps & { isPressed: boolean; isM
 
       // uplift all the nested links so they receive clicks and events (except the LinkOverlay)
       // https://www.sarasoueidan.com/blog/nested-links
-      '& a[href]:not(a[data-blade-component="card-link-overlay"])': {
-        zIndex: 1,
+      [`& a[href]:not(a[data-blade-component="${CARD_LINK_OVERLAY_ID}"])`]: {
+        zIndex: 2,
+        position: 'relative',
+      },
+      [`& button:not(button[data-blade-component="${CARD_LINK_OVERLAY_ID}"])`]: {
```

Comment:
when just adding onClick we change the LinkOverlay to button so this also needs to be set